lld/ELF/Arch/Cheri.cpp

```diff
@@ -835,6 +1147,11 @@ uint64_t CheriCapTableSection::assignIndices(uint64_t startIndex,
     }
   }
 
+  // CheriOS tries to assign indexs in a deterministic way based on the order symbols appear in the input files
```

Comment:
1. Document why, because without knowledge of CheriOS this is just a magic requirement out of nowhere. (Also, it only "tries", so when doesn't it do so, and why is that not a problem?)
2. Would switching CheriBSD to this approach break anything? Ideally we would have the same approach for both.

Reply:
Ive added some extra comment as to why CheriOS does it, but I don't think CheriBSD would benefit. It would deffo slow link times down though, so its probably best to be CheriOS only.